libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1165,6 +1165,22 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
         EMPTY_ELSE_MARKER;
     }
 
+    /* Set the MQTT close callback. */
+    networkStatus = pNewMqttConnection->pNetworkInterface->setCloseCallback( pNetworkConnection,
```

Comment:
This line is causing segfaults in CI, as part of `TEST(MQTT_Unit_API, ConnectParameters)`.

Possible fix:
In `iot_tests_mqtt_api.c`, define function:
```
/**
 * @brief A function for setting the close callback that just returns success.
 */
static IotNetworkError_t _setCloseCallback( IotNetworkConnection_t pConnection,
                                              IotNetworkCloseCallback_t closeCallback,
                                              void * pReceiveContext )
{
    /* Silence warnings about unused parameters. */
    ( void ) pConnection;
    ( void ) closeCallback;
    ( void ) pReceiveContext;

    return IOT_NETWORK_SUCCESS;
}
```
and add the line
`_networkInterface.setCloseCallback = _setCloseCallback;`
to `TEST_SETUP( MQTT_Unit_API )`

Reply:
Good catch, fixed.